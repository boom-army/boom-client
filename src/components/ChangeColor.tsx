import PaletteIcon from '@mui/icons-material/Palette';
import { Stack, Typography } from "@mui/material";
import { ThemeContext } from "../contexts/theme";
import { useState, useContext } from "react";

export const ChangeColor = ({ iconProps, stackProps }: any) => {
  const { theme, setTheme } = useContext(ThemeContext);

  const colors = [
    "#CA2055",
    "#1DA1F2",
    "#FFAD1F",
    "#794BC4",
    "#F45D22",
    "#17BF63",
  ];

  const [currentColor, setCurrentColor] = useState(colors[0]);

  const changeColor = () => {
    let newAccentColor;

    if (colors.indexOf(currentColor) === colors.length - 1) {
      newAccentColor = colors[0];
    } else {
      newAccentColor = colors[colors.indexOf(currentColor) + 1];
    }    

    setTheme({
      ...theme,
      accentColor: newAccentColor,
      pallette: {
        primary: {
          main: newAccentColor,
        }
      }
    });

    setCurrentColor(newAccentColor);
  };

  return (
      <Stack onClick={changeColor} direction="row" {...stackProps} sx={{cursor: "pointer"}}>
        <PaletteIcon sx={iconProps} />
        <Typography variant="body1">Color</Typography>
      </Stack>
  );
};
