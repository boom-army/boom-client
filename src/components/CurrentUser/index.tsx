import { Box, Grid, IconButton } from "@mui/material";
import MenuIcon from "@mui/icons-material/Menu";
import { useToggleDrawer } from "../../hooks";
import { WalletMultiButton } from "@solana/wallet-adapter-material-ui";

export const CurrentUser = () => {
  const toggleDrawer = useToggleDrawer();
  
  return (
    <>
      <Grid
        container
        direction="row"
        justifyContent="flex-start"
        alignItems="center"
      >
        <Box mr={1} display="flex" alignItems="center" >
        <WalletMultiButton />
          <IconButton
            color="primary"
            onClick={toggleDrawer(true)}
            sx={{ ml: 1 }}
          >
            <MenuIcon />
          </IconButton>
        </Box>
      </Grid>
    </>
  );
};
