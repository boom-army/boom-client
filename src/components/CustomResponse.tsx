import React from "react";
import { styled } from "@mui/material/styles";

const Wrapper = styled("div")({
  height: "70vh",
  fontSize: "1.1rem",
  textAlign: "justify",
  display: "flex",
  flexDirection: "column",
  alignItems: "center",
  justifyContent: "center",
  padding: "4rem",
  "@media screen and (max-width: 530px)": {
    fontSize: "1rem",
  },
});

export const CustomResponse = ({ text }: any) => (
  <Wrapper>
    <p>{text}</p>
  </Wrapper>
);
