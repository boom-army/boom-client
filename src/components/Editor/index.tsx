import { AutoFocusPlugin } from "@lexical/react/LexicalAutoFocusPlugin";
import { AutoScrollPlugin } from "@lexical/react/LexicalAutoScrollPlugin";
import { CheckListPlugin } from "@lexical/react/LexicalCheckListPlugin";
import { ClearEditorPlugin } from "@lexical/react/LexicalClearEditorPlugin";
import { ContentEditable } from "@lexical/react/LexicalContentEditable";
import { HashtagPlugin } from "@lexical/react/LexicalHashtagPlugin";
import { LexicalEditor } from "lexical";
import { LinkPlugin } from "@lexical/react/LexicalLinkPlugin";
import { ListPlugin } from "@lexical/react/LexicalListPlugin";
import { OnChangePlugin } from "@lexical/react/LexicalOnChangePlugin";
import { ReactNode, useEffect, useState } from "react";
import { useLexicalComposerContext } from "@lexical/react/LexicalComposerContext";
import { useRef } from "react";

import { RichTextPlugin } from "./plugins/RichTextPlugin";
import { EditorContext } from "./EditorContext";
import { TabFocusPlugin } from "./plugins/TabFocusPlugin";

interface IEditorProps {
  children?: ReactNode;
  hashtagsEnabled?: boolean;
  autoLinkEnabled?: boolean;
  emojisEnabled?: boolean;
  actionsEnabled?: boolean;
  placeholder?: string;
  listMaxIndent?: number;
  initialEditorState?: string;
  isReadOnly?: boolean;
  onChange?: (editorState: string, editorInstance?: LexicalEditor) => void;
}

export const TextEditor = ({
  children,
  hashtagsEnabled = false,
  initialEditorState,
  isReadOnly = false,
  onChange,
}: IEditorProps) => {
  const [editor] = useLexicalComposerContext();
  const [activeEditor, setActiveEditor] = useState(editor);

  const editorStateRef = useRef(null);
  const placeholderComponent = <div>Write something...</div>;
  const scrollRef = useRef(null);

  useEffect(() => {
    editor.setReadOnly(isReadOnly);
  }, []);

  return (
    <EditorContext.Provider
      value={{ initialEditor: editor, activeEditor, setActiveEditor }}
    >
      {children}
      <div className={`editor-container`} ref={scrollRef}>
        <AutoFocusPlugin />
        <ClearEditorPlugin />
        {hashtagsEnabled && <HashtagPlugin />}
        <AutoScrollPlugin scrollRef={scrollRef} />

        <>
          <RichTextPlugin
            contentEditable={<ContentEditable />}
            placeholder={placeholderComponent}
            initialEditorState={initialEditorState}
          />
          <OnChangePlugin
            ignoreInitialChange={false}
            onChange={(editorState) => {
              onChange?.(JSON.stringify(editorState), activeEditor);
              return (editorStateRef.current = editorState);
            }}
          />
          <ListPlugin />
          <CheckListPlugin />
          <LinkPlugin />
          <TabFocusPlugin />
        </>
      </div>
    </EditorContext.Provider>
  );
};
