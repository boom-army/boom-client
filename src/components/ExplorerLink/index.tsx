import React from "react";
import { Typography } from "@mui/material";
import { shortenAddress } from "../../utils/utils";
import { PublicKey } from "@solana/web3.js";

export const ExplorerLink = (props: {
  address: string | PublicKey;
  type: string;
  code?: boolean;
  style?: React.CSSProperties;
  length?: number;
}) => {
  const { type, code } = props;

  const address =
    typeof props.address === "string"
      ? props.address
      : props.address?.toBase58();

  if (!address) {
    return null;
  }

  const length = props.length ?? 4;

  return (
    <a
      href={`https://explorer.solana.com/${type}/${address}`}
      target="_blank"
      title={address}
      style={props.style}
      rel="noreferrer"
    >
      {code ? (
        <Typography>{shortenAddress(address, length)}</Typography>
      ) : (
        shortenAddress(address, length)
      )}
    </a>
  );
};
