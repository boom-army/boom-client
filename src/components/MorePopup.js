import React, { useContext } from "react";
import { styled } from "@mui/material/styles";
import Popup from "reactjs-popup";
import ToggleTheme from "./ToggleTheme";
import Logout from "./Auth/Logout";
import { ChangeColor } from "./ChangeColor";
import MoreHorizIcon from "@mui/icons-material/MoreHoriz";
import { Stack, Typography } from "@mui/material";
import { ThemeContext } from "../contexts/theme";

const Wrapper = styled("div")({
  ".btn": {
    cursor: "pointer",
  },
});

const MoreBtn = React.forwardRef(
  ({ open, iconProps, stackProps, ...props }, ref) => {
    return (
      <div ref={ref} {...props}>
        <Stack direction="row" {...stackProps} sx={{ cursor: "pointer" }}>
          <MoreHorizIcon sx={iconProps} />
          <Typography
            variant="body1"
            display={{ xs: "none", sm: "none", md: "block", lg: "block" }}
          >
            More
          </Typography>
        </Stack>
      </div>
    );
  }
);

export const MorePopUp = ({ iconProps, stackProps }) => {
  const { theme } = useContext(ThemeContext);

  const contentStyle = {
    width: "160px",
    background: theme.background,
    borderRadius: "6px",
    border: `1px solid ${theme.tertiaryColor}`,
    boxShadow: theme.bs1,
    padding: "1rem 1rem 0 1rem",
  };

  const overlayStyle = {
    background: "none",
  };

  return (
    <Wrapper>
      <Popup
        className="btn"
        trigger={(open) => (
          <MoreBtn open={open} iconProps={iconProps} stackProps={stackProps} />
        )}
        position="top left"
        closeOnDocumentClick
        contentStyle={contentStyle}
        overlayStyle={overlayStyle}
        arrow={false}
      >
        <ToggleTheme />
        <ChangeColor />
        <Logout />
      </Popup>
    </Wrapper>
  );
};
