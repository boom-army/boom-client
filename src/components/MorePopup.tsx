import React, { useContext } from "react";
import { styled } from "@mui/material/styles";
import Popup from "reactjs-popup";
import ToggleTheme from "./ToggleTheme";
import Logout from "./Auth/Logout";
import { ChangeColor } from "./ChangeColor";
import MoreHorizIcon from "@mui/icons-material/MoreHoriz";
import { Stack, Typography } from "@mui/material";
import { ThemeContext } from "../contexts/theme";

const Wrapper = styled("div")({
  ".btn": {
    cursor: "pointer",
  },
});

const StyledPopup = styled(Popup)({
  "@media (max-width: 899px)": {
    position: "absolute",
    right: "13em",
    top: "3em",
    zIndex: 9999,
  },
});

interface Props {
  open?: boolean;
  iconProps?: any;
  stackProps?: any;
}

const MoreBtn = React.forwardRef(
  (
    { open, iconProps, stackProps, ...props }: Props,
    ref: React.LegacyRef<HTMLDivElement> | undefined
  ) => {
    return (
      <div ref={ref} {...props}>
        <Stack direction="row" {...stackProps} sx={{ cursor: "pointer" }}>
          <MoreHorizIcon sx={iconProps} />
          <Typography
            variant="body1"
            display={{ xs: "none", sm: "none", md: "block", lg: "block" }}
          >
            More
          </Typography>
        </Stack>
      </div>
    );
  }
);

export const MorePopUp = ({ iconProps, stackProps }: Props) => {
  const { theme } = useContext(ThemeContext);

  const contentStyle = {
    width: "160px",
    background: theme.background,
    borderRadius: "6px",
    border: `1px solid ${theme.tertiaryColor}`,
    boxShadow: theme.bs1,
    padding: "1rem 1rem 0 1rem",
  };

  const overlayStyle = {
    background: "none",
  };

  return (
    <Wrapper>
      <StyledPopup
        className="btn"
        trigger={(open) => (
          <MoreBtn open={open} iconProps={iconProps} stackProps={stackProps} />
        )}
        position="top left"
        closeOnDocumentClick
        contentStyle={contentStyle}
        overlayStyle={overlayStyle}
        arrow={false}
      >
        <ToggleTheme />
        <ChangeColor />
        <Logout />
      </StyledPopup>
    </Wrapper>
  );
};
