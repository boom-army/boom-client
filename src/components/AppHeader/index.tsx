import { useCallback, useEffect, useMemo, useContext } from "react";
import base58 from "bs58";
import {
  AppBar,
  Container,
  Grid,
  Toolbar,
} from "@mui/material";
import { CurrentUser } from "../CurrentUser";
import { PUBLIC_ADDRESS, LOGIN_REGISTER } from "../../queries/auth";
import { USER_FOLLOW } from "../../queries/follow";
import { Theme } from "../../contexts/theme";
import { UserContext } from "../../contexts/user";
import { useMutation } from "@apollo/client";
import { useSnackbar } from "../../contexts/snackbar";
import { useWallet } from "@solana/wallet-adapter-react";
import { FeedDocument } from "../../generated/graphql";
import { Box } from "@mui/system";
import { Link } from "react-router-dom";
import { RoutePath } from "../../constants";

export const AppHeader = () => {
  const { connected, wallet, publicKey, signMessage } = useWallet();
  const { enqueueSnackbar } = useSnackbar();
  const { setUser } = useContext(UserContext);

  const token = localStorage.getItem("token");
  const headerImg =
    localStorage.getItem(Theme.StorageTag) === Theme.Light
      ? "/assets/boom-logo-dark.png"
      : "/assets/boom-logo-light.png";

  const [getNonce] = useMutation(PUBLIC_ADDRESS);
  const [setLogin] = useMutation(LOGIN_REGISTER, {
    refetchQueries: [{ query: FeedDocument }, { query: USER_FOLLOW }],
    onCompleted({ loginRegister }) {
      if (localStorage) {
        setUser(loginRegister.user);
        localStorage.setItem("token", loginRegister.token);
      }
    },
  });

  const walletPublicKey = useMemo(() => publicKey?.toBase58(), [publicKey]);

  const signin = useCallback(async () => {
    if (!walletPublicKey) {
      enqueueSnackbar("Wallet not connected!", { variant: "error" });
      return;
    }

    if (!signMessage) {
      enqueueSnackbar("Wallet does not support message signing!", {
        variant: "error",
      });
      return;
    }

    try {
      const {
        data: { address },
      } = await getNonce({
        variables: { publicAddress: walletPublicKey },
      });
      if (address.hasPublicAddress) {
        const data = new TextEncoder().encode(address.user.nonce);
        const signature = await signMessage(data);

        await setLogin({
          variables: {
            publicAddress: walletPublicKey,
            signature: base58.encode(signature),
          },
        });
        window.location.reload();
        enqueueSnackbar(
          `Wallet ${walletPublicKey} connected to account. Happy posting.`,
          { variant: "success" }
        );
      } else {
        await setLogin({ variables: { publicAddress: walletPublicKey } });
        enqueueSnackbar(`Wallet ${walletPublicKey} created for account.`, {
          variant: "success",
        });
      }
    } catch (error) {
      enqueueSnackbar(`Error connecting: ${error}`, { variant: "error" });
    }
  }, [signMessage, getNonce, setLogin, walletPublicKey, enqueueSnackbar]);

  useEffect(() => {
    if (wallet && !token && connected) signin();
  }, [wallet, signin, token, connected]);

  // const logout = () => {
  //   localStorage.clear();
  //   window.location.replace('/');
  // };

  return (
    <AppBar
      position="static"
      sx={{
        borderBottom: (t) => `1px solid ${t.palette.divider}`,
      }}
    >
      <Toolbar variant="dense">
        {/* Remove this when we go full width */}
        <Container maxWidth="lg">
          <Grid container>
            <Grid item xs={12}>
              <Box display="flex" justifyContent="space-between">
                <Box mr={1} display="flex" alignItems="center">
                  <Link
                    to={RoutePath.HOME}
                    style={{ display: "flex", alignItems: "center" }}
                  >
                    <img src={headerImg} alt="Boom" height={20} />
                  </Link>
                </Box>
                <Box mt={0.5}>
                  <CurrentUser />
                </Box>
              </Box>
            </Grid>
          </Grid>
        </Container>
      </Toolbar>
    </AppBar>
  );
};
