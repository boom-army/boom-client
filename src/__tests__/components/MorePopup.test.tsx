import React from "react";
import { MorePopUp } from "../../components/MorePopup";
import { act, fireEvent, render, screen } from "@testing-library/react";
import { MemoryRouter } from "react-router-dom";
import { createSerializer } from '@emotion/jest'

expect.addSnapshotSerializer(createSerializer(
  {
    classNameReplacer(className, index) {
      return `css-${index}`
    },
      // includeStyles: false 
  }
));


describe("<MorePopup/> component :", () => {
  test("display <MorePopup/> component ", async () => {
    let rendered = render(<MorePopUp />, { wrapper: MemoryRouter });
    expect(rendered).toMatchSnapshot();
  });
});

test("Assertion testing of <MorePopup/>  component", () => {
  render(<MorePopUp />, { wrapper: MemoryRouter });
  expect(screen.getByText("More")).toBeInTheDocument();
});
