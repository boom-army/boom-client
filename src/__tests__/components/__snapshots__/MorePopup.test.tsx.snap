// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<MorePopup/> component : display <MorePopup/> component  1`] = `
Object {
  "asFragment": [Function],
  "baseElement": .css-0 .btn {
  cursor: pointer;
}

.css-1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  cursor: pointer;
}

.css-2 {
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  width: 1em;
  height: 1em;
  display: inline-block;
  fill: currentColor;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  -webkit-transition: fill 200ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  transition: fill 200ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  font-size: 1.5rem;
}

.css-3 {
  margin: 0;
  font-family: "Roboto","Helvetica","Arial",sans-serif;
  font-weight: 400;
  font-size: 1rem;
  line-height: 1.5;
  letter-spacing: 0.00938em;
}

@media (min-width:0px) {
  .css-3 {
    display: none;
  }
}

@media (min-width:600px) {
  .css-3 {
    display: none;
  }
}

@media (min-width:900px) {
  .css-3 {
    display: block;
  }
}

@media (min-width:1200px) {
  .css-3 {
    display: block;
  }
}

<body>
    <div>
      <div
        class="css-0"
      >
        <div
          aria-describedby="popup-1"
        >
          <div
            class="css-1"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-2"
              data-testid="MoreHorizIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M6 10c-1.1 0-2 .9-2 2s.9 2 2 2 2-.9 2-2-.9-2-2-2zm12 0c-1.1 0-2 .9-2 2s.9 2 2 2 2-.9 2-2-.9-2-2-2zm-6 0c-1.1 0-2 .9-2 2s.9 2 2 2 2-.9 2-2-.9-2-2-2z"
              />
            </svg>
            <p
              class="MuiTypography-root MuiTypography-body1 css-3"
            >
              More
            </p>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": .css-0 .btn {
  cursor: pointer;
}

.css-1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  cursor: pointer;
}

.css-2 {
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  width: 1em;
  height: 1em;
  display: inline-block;
  fill: currentColor;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  -webkit-transition: fill 200ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  transition: fill 200ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  font-size: 1.5rem;
}

.css-3 {
  margin: 0;
  font-family: "Roboto","Helvetica","Arial",sans-serif;
  font-weight: 400;
  font-size: 1rem;
  line-height: 1.5;
  letter-spacing: 0.00938em;
}

@media (min-width:0px) {
  .css-3 {
    display: none;
  }
}

@media (min-width:600px) {
  .css-3 {
    display: none;
  }
}

@media (min-width:900px) {
  .css-3 {
    display: block;
  }
}

@media (min-width:1200px) {
  .css-3 {
    display: block;
  }
}

<div>
    <div
      class="css-0"
    >
      <div
        aria-describedby="popup-1"
      >
        <div
          class="css-1"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-2"
            data-testid="MoreHorizIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M6 10c-1.1 0-2 .9-2 2s.9 2 2 2 2-.9 2-2-.9-2-2-2zm12 0c-1.1 0-2 .9-2 2s.9 2 2 2 2-.9 2-2-.9-2-2-2zm-6 0c-1.1 0-2 .9-2 2s.9 2 2 2 2-.9 2-2-.9-2-2-2z"
            />
          </svg>
          <p
            class="MuiTypography-root MuiTypography-body1 css-3"
          >
            More
          </p>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
